src/commands/plugins.rs

```diff
@@ -242,11 +243,72 @@ impl Upgrade {
 
         if self.all {
             self.upgrade_all(manifests_dir).await
+        } else if self.name.is_none()
+            && self.local_manifest_src.is_none()
+            && self.remote_manifest_src.is_none()
+        {
+            // Default behavior (multiselect)
+            self.upgrade_multiselect().await
         } else {
             self.upgrade_one().await
         }
     }
 
+    // Multiselect plugin upgrade experience
+    async fn upgrade_multiselect(self) -> Result<()> {
+        let catalogue_plugins = list_catalogue_plugins().await?;
+        let installed_plugins = list_installed_plugins()?;
+
+        let installed_in_catalogue: Vec<_> = installed_plugins
+            .into_iter()
+            .filter(|installed| {
+                catalogue_plugins.iter().any(|catalogue| {
+                    installed.manifest == catalogue.manifest
+                        && matches!(catalogue.compatibility, PluginCompatibility::Compatible)
+                })
+            })
+            .collect();
+
+        eprintln!(
+            "Select plugins to upgrade. Use Space to select/deselect and Enter to confirm selection."
+        );
+        let selected_indexes = match dialoguer::MultiSelect::new()
+            .items(&installed_in_catalogue)
```

Comment:
This looks like it currently displays _all_ plugins installed from the catalogue, even those without an eligible upgrade.

Reply:
You're correct. What I tried to do to check which plugins are eligible for an upgrade was to call `get_manifest()` with the latest version (using `manifest_location(None)`) for each plugin. If the function returns successfully, indicating an update is available, I push it to the new `eligible_plugins` vector with the descriptor and the new manifest to have both the old and latest versions.